server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -421,61 +421,43 @@ public static Builder builder(GatekeeperOptions gatekeeperOptions) {
     /**
      * Defines which selectors should be used by default for an index operation in the event that no selectors are provided.
      */
-    public record SelectorOptions(EnumSet<IndexComponentSelector> defaultSelectors) implements Writeable {
+    public record SelectorOptions(IndexComponentSelector defaultSelector) implements Writeable {
 
-        public static final SelectorOptions DATA_AND_FAILURE = new SelectorOptions(
-            EnumSet.of(IndexComponentSelector.DATA, IndexComponentSelector.FAILURES)
-        );
-        public static final SelectorOptions ONLY_DATA = new SelectorOptions(EnumSet.of(IndexComponentSelector.DATA));
-        public static final SelectorOptions ONLY_FAILURES = new SelectorOptions(EnumSet.of(IndexComponentSelector.FAILURES));
+        public static final SelectorOptions ALL_APPLICABLE = new SelectorOptions(IndexComponentSelector.ALL_APPLICABLE);
+        public static final SelectorOptions DATA = new SelectorOptions(IndexComponentSelector.DATA);
+        public static final SelectorOptions FAILURES = new SelectorOptions(IndexComponentSelector.FAILURES);
         /**
          * Default instance. Uses <pre>::data</pre> as the default selector if none are present in an index expression.
          */
-        public static final SelectorOptions DEFAULT = ONLY_DATA;
+        public static final SelectorOptions DEFAULT = DATA;
 
         public static SelectorOptions read(StreamInput in) throws IOException {
-            return new SelectorOptions(in.readEnumSet(IndexComponentSelector.class));
+            if (in.getTransportVersion().before(TransportVersions.INTRODUCE_ALL_APPLICABLE_SELECTOR)) {
+                EnumSet<IndexComponentSelector> set = in.readEnumSet(IndexComponentSelector.class);
+                if (set.isEmpty() || set.size() == 2) {
+                    return SelectorOptions.ALL_APPLICABLE;
```

Comment:
Can you add `assert set.contains(IndexComponentSelector.DATA) && set.contains(IndexComponentSelector.FAILURES) : "…"` here, just for a sanity check in case the enum set ever expands?